code/datums/gamemode/role/ninja.dm

```diff
@@ -379,6 +387,10 @@
 	else
 		to_chat(user,"<span class='notice'>The glove's power flows into your weapon. It will be ready in [round((T.cooldown - world.time)/10)] seconds.</span>")
 
+	var/datum/role/ninja/N = user.mind.GetRole(NINJA)
+	N.times_charged_sword++
```

Comment:
this applies to everything, but why increment it on the role with a new var instead of just going on through to get the stat datum and directly incrementing it there?

Reply:
Honestly, I can't remember. I think I was trying to be minimally invasive to faction/role datums logic itself, but I've accomplished the opposite here, in the end. I'll probably change it back unless I can remember why I did this.

Reply:
Reviewed my code. Already posted in Discord but I'll put it here as well. Basically I did this so that within faction/role logic, you wouldn't have to type check a stat datum to access its variables every time. The statistics are instead stored on the role/faction datum itself, then on `New()` the faction/role stat specific datum just grabs the data already on the game's role/faction. No type checks involved.

Is it worth keeping? Is one better than the other? I have no clue!